packages/core/src/index.ts

```diff
@@ -85,26 +95,45 @@ export const create = (config: Config): VoiceCompass => {
     },
   });
 
-  let stepId: string | null = null;
+  let _stepId: string | null = null;
 
-  const updateStep = (stepData: StepData) => {
+  const updateStep = <T>({
+    stepId,
+    journeyId = config.journeyId,
+    end,
+    escalate,
+    payload,
+  }: StepData<T>) => {
     // skip step if the previous stepId is the same as the current stepId
-    if (stepData.stepId === stepId) {
+    if (stepId === _stepId) {
       return;
     }
 
-    client
-      .post("/track", {
-        ...stepData,
-        contactId: config.contactId,
-        botId: config.botId,
-        journeyId: stepData.journeyId || config.journeyId,
-        voice: config.voice,
-        language: config.language,
-      })
-      .then(() => {
-        stepId = stepData.stepId;
-      });
+    const _payload = {
+      stepId,
+      ...(typeof end === "boolean" ? { end } : {}),
+      ...(typeof escalate === "boolean" ? { escalate } : {}),
+      ...(typeof payload === "object" &&
```

Comment:
object checking is always a pain

Reply:
Different PR please. But also what about `{ end: Boolean(end), escalate: Boolean(escalate) ... }`